src/schemas/__tests__/__snapshots__/Entity.test.js.snap

```diff
@@ -92,6 +92,100 @@ Object {
 }
 `;
 
+exports[`EntitySchema denormalization denormalizes recursive dependencies 1`] = `
+Object {
+  "draftedBy": Object {
+    "id": "456",
+    "reports": Array [
+      [Circular],
+    ],
+    "role": "manager",
+  },
+  "id": "123",
+  "publishedBy": Object {
+    "id": "456",
+    "reports": Array [
+      [Circular],
+    ],
+    "role": "manager",
+  },
+  "title": "Weekly report",
+}
+`;
+
+exports[`EntitySchema denormalization denormalizes recursive dependencies 2`] = `
+Object {
+  "draftedBy": Object {
+    "id": "456",
+    "reports": Array [
+      Object {
+        "draftedBy": "456",
+        "id": "123",
+        "publishedBy": "456",
+        "title": "Weekly report",
+      },
+    ],
+    "role": "manager",
+  },
+  "id": "123",
+  "publishedBy": Object {
+    "id": "456",
+    "reports": Array [
+      Object {
```

Comment:
Why do Immutables go an extra level deep instead of stopping at the first duplication? I'd expect this to just be `"123"` instead of a repeat of the original object.